packages/p2p/src/components/my-ads/edit-ad-summary.jsx

```diff
@@ -6,18 +6,43 @@ import { Text } from '@deriv/components';
 import { buy_sell } from 'Constants/buy-sell';
 import { Localize } from 'Components/i18next';
 import { useStores } from 'Stores';
+import { roundOffDecimal, setDecimalPlaces } from 'Utils/format-value.js';
 
-const EditAdSummary = ({ offer_amount, price_rate, type }) => {
+const EditAdSummary = ({ market_feed, offer_amount, price_rate, type }) => {
     const { general_store } = useStores();
     const { currency, local_currency_config } = general_store.client;
+    const decimal_places = setDecimalPlaces(market_feed, 6);
 
     const display_offer_amount = offer_amount ? formatMoney(currency, offer_amount, true) : '';
-    const display_price_rate = price_rate ? formatMoney(local_currency_config.currency, price_rate, true) : '';
-    const display_total =
-        offer_amount && price_rate ? formatMoney(local_currency_config.currency, offer_amount * price_rate, true) : '';
+    let display_price_rate = '';
+    let display_total = '';
+
+    if (market_feed && price_rate) {
+        display_price_rate = formatMoney(
+            local_currency_config.currency,
+            roundOffDecimal(parseFloat(market_feed * (1 + price_rate / 100)), decimal_places),
+            true,
+            decimal_places
+        );
+    } else if (price_rate) {
+        display_price_rate = formatMoney(local_currency_config.currency, price_rate, true);
+    }
+
+    if (market_feed && offer_amount && price_rate) {
```

Comment:
You're repeating the check in the `else if (offer_amount && price_rate)` condition on line 37.  You can wrap this in a check for offer_amount and price_rate.